Source/ParameterEncoding.swift

```diff
@@ -161,16 +193,16 @@ public struct URLEncoding: ParameterEncoding {
             }
         } else if let array = value as? [Any] {
             for value in array {
-                components += queryComponents(fromKey: "\(key)[]", value: value)
+                components += queryComponents(fromKey: encodeArrayKey(key), value: value)
             }
         } else if let value = value as? NSNumber {
             if value.isBool {
-                components.append((escape(key), escape((value.boolValue ? "1" : "0"))))
+                components.append((escape(key), escape(encodeBoolValue(value.boolValue))))
```

Comment:
With `encode` on `BoolEncoding`, this becomes `boolEncoding.encode(value.boolValue)`.